### PR TITLE
Type checker fixes

### PR DESCRIPTION
**Pull request by richardzilincikPantheon**

### CODE REVIEW

File: bin/create_config.py [hunk 1]
```diff
@@ -4,6 +4,6 @@
 
 def create_config(config_path=os.environ['YANGCATALOG_CONFIG_PATH']):
     config = configparser.ConfigParser()
-    config._interpolation = configparser.ExtendedInterpolation()
+    config._interpolation = configparser.ExtendedInterpolation() # pyright: ignore
```
Comment (SlavomirMazurPantheon) [inline bin/create_config.py:7]: Use `parser = ConfigParser(interpolation=ExtendedInterpolation())` instead?

File: bin/get_config.py [hunk 2]
```diff
@@ -34,7 +34,7 @@
     args = parser.parse_args()
 
     config = configparser.ConfigParser()
-    config._interpolation = configparser.ExtendedInterpolation()
+    config._interpolation = configparser.ExtendedInterpolation() # pyright: ignore
```
Comment (SlavomirMazurPantheon) [inline bin/get_config.py:37]: `parser = ConfigParser(interpolation=ExtendedInterpolation())`

File: bin/symd.py [hunk 3]
```diff
@@ -143,19 +143,19 @@ def get_yang_modules(yfiles, tag):
                     rev = None
                 # IF we already have a module with a lower revision, replace it now
                 try:
-                    en = G.node[name]
+                    en = G.nodes[name]
```
Comment (SlavomirMazurPantheon) [inline bin/symd.py:146]: Object of type "(data: bool = False) -> list[tuple[Unknown, Unknown]]" is not subscriptable

**Comment by SlavomirMazurPantheon** [comment]
Please check symd.py script once again, there is an error:
```
Traceback (most recent call last):
  File "/./sdo_analysis/bin/symd.py", line 461, in <module>
    init(args.rfc_repos, args.draft_repos, recurse=args.recurse)
  File "/./sdo_analysis/bin/symd.py", line 375, in init
    get_yang_modules(rfc_yang_files, RFC_TAG)
  File "/./sdo_analysis/bin/symd.py", line 149, in get_yang_modules
    en = G.nodes[name]
TypeError: 'method' object is not subscriptable
```

File: bin/create_config.py (hunk 1 above)
Comment (richardzilincikPantheon) [reply bin/create_config.py:7]: Ah, yeah that looks like the proper way to do it.

**Pull request merged**